### PR TITLE
gpui: Add support for polychrome SVGs

### PR DESCRIPTION
**Pull request by aminya**
This allows the SVG renderer in Zed to respect the colors specified in the SVG file and not override them with the default monochrome color. This opens up the possibility of creating colorful file icons.

Related issues:

- #8843 
- #4609
- #6150
- #5525 

Release Notes:

- N/A

### CODE REVIEW

File: crates/ui/src/components/icon.rs [hunk 1]
```diff
@@ -496,12 +496,16 @@ impl Icon {
 
 impl RenderOnce for Icon {
     fn render(self, cx: &mut WindowContext) -> impl IntoElement {
-        svg()
+        let svg = svg()
```
Comment (maxdeviant) [inline crates/ui/src/components/icon.rs:499]: We don't want to make any changes to the `Icon` yet, as right now the intent in Zed is for all icons to by monochrome.

Comment (aminya) [reply crates/ui/src/components/icon.rs:499]: This PR doesn't change the icons. It just allows the renderer to work correctly and respect the colours. All the default icons are still monochrome.

Comment (maxdeviant) [reply crates/ui/src/components/icon.rs:499]: But it's still making changes to the `Icon` component, which could have unintended effects.

I would prefer if we kept the scope limited just to GPUI right now.

Comment (aminya) [reply crates/ui/src/components/icon.rs:499]: Well, I am not sure that the current defaults are good. Currently, GPUI makes all SVGs monochrome. This is a huge surprise to anyone who has worked with a  UI framework. 

This PR enables the correct behaviour using `with_polychrome`, but if I was going to write this from scratch, I would reverse the default. Respect the colours, unless explicitly requested by the user.

So, if you prefer, I can do this, and explicitly force all Icons to be monochrome until we decide what should be done regarding the icons.

Comment (kyjus25) [reply crates/ui/src/components/icon.rs:499]: So like... maybe I'm naive, but why does the framework need to override the colors? Why can't the SVG itself just be monochrome? Sounds to me like a utility abstraction that is doing more harm than good based on how much the community seems to _really_ want colorful icons.

**Comment by 0xFlo** [comment]
I still don't see colorful icons in Zed and have no way to configure them. Not sure how this is solving anything if you then go and say we want to keep monochrome icons.

**Comment by maxdeviant** [comment]
The functionality in this PR doesn't seem to work, based on my testing.

Supporting polychrome SVGs in GPUI is something that we would be interested in supporting, but we're not at a point where we're ready to make use of that in Zed itself (until we start supporting third-party icons via extensions).

**Pull request closed without merging**